### PR TITLE
Add references for cloud resource organization

### PR DESCRIPTION
**Pull request by bderusha**
# Pull Request Template

## What are you trying to address

- #687 

## Description of new changes

- Adding a reference guide to highlight cloud resource organization best practices.

## For all pull requests

- [x] Link to the issue you are solving (so it gets closed)
- [ ] Label the pull request with the appropriate area(s)
- [ ] Assign potential reviewers (you may also want to contact them on Teams to ensure timely reviews)
- [ ] Assign the project - Engineering Playbook Backlog
- [ ] Assign the pull request to yourself

### CODE REVIEW

**Comment by bderusha** [comment]
CC @TessFerrandez

File: docs/design/design-patterns/cloud-resource-design-reference/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Cloud Resource Design Reference
+
+As cloud usage scales, considerations for subscription design and resource naming/tagging conventions have an impact on governance, operations management, and adoption patterns.
```
Comment (shiranr) [inline docs/design/design-patterns/cloud-resource-design-reference/README.md:3]: I would consider creating a new folder for cloud resource design and put this file inside. 
I think there can be more design patterns for cloud env.

**Review by meulta — COMMENTED** [review]
Thank you for your contribution!!!

Quick question: If you do not recommend implementing these guidelines for their own sake, what is the correct guidance then? Can you add a bit more direction on the right approach to considering them?

**Comment by bderusha** [comment]
> Thank you for your contribution!!!
> 
> Quick question: If you do not recommend implementing these guidelines for their own sake, what is the correct guidance then? Can you add a bit more direction on the right approach to considering them?

@meulta The point was meant to emphasize the importance of understanding *why* something is a best-practice so that one can apply critical thinking skills to ensure that it is relevant (or at least non-harmful) to the specific context it's being utilized in.  And that in a pre-existing environment one should be even more critical of the impact of any change.  [Organizing Dev/Test/Production Workloads](https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/enterprise-scale/faq#how-do-we-handle-devtestproduction-workload-landing-zones-in-enterprise-scale-architecture) offers a great example of understanding the recommendation and how it may not apply to all scenarios.

All that said, I agree that the statement is a bit ambiguous in delivering that point :) I will push up a reworded version.

**Review by TessFerrandez — APPROVED** [review]

**Review by shiranr — APPROVED** [review]

**Pull request merged**